rllib/algorithms/algorithm.py

```diff
@@ -2899,7 +2908,7 @@ def get_checkpointable_components(self) -> List[Tuple[str, "Checkpointable"]]:
     @override(Checkpointable)
     def get_ctor_args_and_kwargs(self) -> Tuple[Tuple, Dict[str, Any]]:
         return (
-            (self.config,),  # *args,
+            (self.config.get_state(),),  # *args,
```

Comment:
Perfect!

Reply:
Btw, we should probably do the same everywhere else `config` is part of the c'tor args/kwargs: Learner, LearnerGroup, and EnvRunner.

Reply:
Then, we also have to make sure their c'tors also accept config states(!), not just AlgorithmConfig objects (the same as how Algorithm does it now).